File: app/src/main/java/org/michaelbel/tjgram/data/entity/BaseResult.java
package org.michaelbel.tjgram.data.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class BaseResult<T> implements Serializable {
    @Expose @SerializedName("result") public T result;
    @Expose @SerializedName("message") public String message;
}

File: app/src/main/java/org/michaelbel/tjgram/data/entities/BaseResult.kt
package org.michaelbel.tjgram.data.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

open class BaseResult<T>: Serializable {
    @Expose @SerializedName("result") var result: T? = null
    @Expose @SerializedName("message") var message: String? = null
}